tests/unit/test_pull_request.py

```diff
@@ -0,0 +1,364 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Unit tests for git."""
+
+# Need access to protected functions for testing
+# pylint: disable=protected-access
+
+from pathlib import Path
+from unittest import mock
+
+import pytest
+from git.exc import GitCommandError
+from git.repo import Repo
+from github import Github
+from github.GithubException import GithubException
+from github.PullRequest import PullRequest
+from github.Repository import Repository
+
+from src import pull_request
+from src.exceptions import InputError, RepositoryClientError
+from src.pull_request import RepositoryClient
+
+from .helpers import assert_substrings_in_string
+
+
+def test___init__(repository: Repo, mock_github_repo: Repository):
```

Comment:
`test_repo___init__` to indicate it is the init for the repo class? Would that help? Similar for other test functions in this file targeted at the repo class

Reply:
added a prefix `repository_client` to tests that are for the RepositoryClient class.
Thank you for pointing that out!